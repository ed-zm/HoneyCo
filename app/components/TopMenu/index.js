/**
 *
 * TopMenu
 *
 */

import React from 'react'
import { Link } from 'react-router'
import { Container, Collapse, Nav, Navbar, NavbarBrand, NavbarToggler, NavItem, NavLink } from 'reactstrap'
import styled from 'styled-components'
import { HoneyCoBlueRGBA, HoneyCoYellow } from '../../constants'

import HorizLogoLockup from './static/logo-horiz-white@0.5x.png'

const StyledNavbar = styled(Navbar)`
  background-color: ${(props) => props.opacity ? HoneyCoBlueRGBA(1) : HoneyCoBlueRGBA(0)};
  font-size: 26px;
  font-weight: bolder;
  position: fixed;
  padding: 10px 0px;
  top: 0;
  left: 0;
  z-index: 10;
  width: 100%;
  transition: background-color 1s;
`

const StyledNavItem = styled(NavItem)`
  padding: 0em .2em;
  font-size: 26px;
  font-weight: lighter;
  &:hover{
    border-bottom: 5px solid white;
  }
`

const StyledCollapse = styled(Collapse)`
  justify-content: flex-end;
`

const BiggerNavbarToggler = styled(NavbarToggler)`
  .navbar-toggler-icon {
    height: 2em;
    width: 2em;
  }
`

const WhiteNavLink = styled(NavLink)`
  color: white !important;
  &.active {
    color: ${HoneyCoYellow} !important;
  }
`

class TopMenu extends React.Component { // eslint-disable-line react/prefer-stateless-function
  constructor (props) {
    super(props)

    this.toggle = this.toggle.bind(this)
    this.changeBackground = this.changeBackground.bind(this)
    this.state = {
      isOpen: false,
      opacity: false,
    }
  }
  componentWillMount () {
    window.addEventListener('scroll', this.changeBackground)
  }
  toggle () {
    this.setState({
      isOpen: !this.state.isOpen,
    })
  }
  changeBackground () {
    if (window.scrollY !== 0) {
      this.setState({ opacity: true })
    } else {
      this.setState({ opacity: false })
    }
  }

  render () {
    const { menuItems } = this.props

    return (
      <StyledNavbar toggleable = "md" opacity = {this.state.opacity} light>
        <Container>
          <BiggerNavbarToggler right onClick={this.toggle} />
          <NavbarBrand tag={Link} to="/">
            <img src={HorizLogoLockup} width={215} height={46} alt="HoneyCo Homes" />
          </NavbarBrand>
          <StyledCollapse isOpen={this.state.isOpen} navbar>
            <Nav className="ml-2" navbar>
              {
                menuItems.map((item) =>
                  <StyledNavItem key={item.name}>
                    <WhiteNavLink tag={Link} to={item.route} activeClassName="active">{item.name}</WhiteNavLink>
                  </StyledNavItem>,
                )
              }
            </Nav>
          </StyledCollapse>
        </Container>
      </StyledNavbar>
    )
  }
}

TopMenu.propTypes = {
  menuItems: React.PropTypes.array,
  ctaButton: React.PropTypes.object,
}

TopMenu.defaultProps = {
  menuItems: [
    { name: 'Who its for', route: '/who-its-for' },
    { name: 'How it works', route: '/how-it-works' },
    { name: 'Shop', route: '/shop' },
    { name: 'Account', route: '/account' }],
  ctaButton: { route: '/', name: 'I\'ve decided' },
}

export default TopMenu
