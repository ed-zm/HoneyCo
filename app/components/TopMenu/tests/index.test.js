import React from 'react'
import { shallow } from 'enzyme'
import TopMenu from '../index'

describe('<TopMenu />', () => {
  it('It should render the TopMenu Component', () => {
    const wrapper = shallow(<TopMenu/>)
    expect(wrapper).toMatchSnapshot()
  })
})
