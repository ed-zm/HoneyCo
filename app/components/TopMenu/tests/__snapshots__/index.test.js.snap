// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopMenu /> It should render the TopMenu Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(jt)
    light={true}
    toggleable="md"
>
    <gt
        tag="div"
    >
        <Styled(Wt)
            onClick={[Function]}
            right={true}
        />
        <St
            tag={[Function]}
            to="/"
        >
            <img
                alt="HoneyCo Homes"
                height={46}
                src="IMAGE_MOCK"
                width={215}
            />
        </St>
        <Styled(ei)
            isOpen={false}
            navbar={true}
        >
            <Gt
                className="ml-2"
                navbar={true}
                tag="ul"
            >
                <Styled($t)
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                >
                    <Styled(ei)
                        activeClassName="active"
                        tag={[Function]}
                        to="#"
                    >
                        Who its for
                        <styled.div
                            display="none"
                        >
                            <styled.ul>
                                <styled.li>
                                    <Styled(Link)
                                        to="/resident"
                                    >
                                        Resident
                                    </Styled(Link)>
                                </styled.li>
                                <styled.li>
                                    <Styled(Link)
                                        to="/caregiver"
                                    >
                                        Caregiver
                                    </Styled(Link)>
                                </styled.li>
                                <styled.li>
                                    <Styled(Link)
                                        to="/partnerships"
                                    >
                                        Partnerships
                                    </Styled(Link)>
                                </styled.li>
                            </styled.ul>
                        </styled.div>
                    </Styled(ei)>
                </Styled($t)>
                <Styled($t)
                    onMouseEnter=""
                    onMouseLeave=""
                >
                    <Styled(ei)
                        activeClassName="active"
                        tag={[Function]}
                        to="/how-it-works"
                    >
                        How it works
                        
                    </Styled(ei)>
                </Styled($t)>
                <Styled($t)
                    onMouseEnter=""
                    onMouseLeave=""
                >
                    <Styled(ei)
                        activeClassName="active"
                        tag={[Function]}
                        to="/shop"
                    >
                        Shop
                        
                    </Styled(ei)>
                </Styled($t)>
                <Styled($t)
                    onMouseEnter=""
                    onMouseLeave=""
                >
                    <Styled(ei)
                        activeClassName="active"
                        tag={[Function]}
                        to="/account"
                    >
                        Account
                        
                    </Styled(ei)>
                </Styled($t)>
            </Gt>
        </Styled(ei)>
    </gt>
</Styled(jt)>,
  "nodes": Array [
    <Styled(jt)
      light={true}
      toggleable="md"
>
      <gt
            tag="div"
      >
            <Styled(Wt)
                  onClick={[Function]}
                  right={true}
            />
            <St
                  tag={[Function]}
                  to="/"
            >
                  <img
                        alt="HoneyCo Homes"
                        height={46}
                        src="IMAGE_MOCK"
                        width={215}
                  />
            </St>
            <Styled(ei)
                  isOpen={false}
                  navbar={true}
            >
                  <Gt
                        className="ml-2"
                        navbar={true}
                        tag="ul"
                  >
                        <Styled($t)
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                        >
                              <Styled(ei)
                                    activeClassName="active"
                                    tag={[Function]}
                                    to="#"
                              >
                                    Who its for
                                    <styled.div
                                          display="none"
                                    >
                                          <styled.ul>
                                                <styled.li>
                                                      <Styled(Link)
                                                            to="/resident"
                                                      >
                                                            Resident
                                                      </Styled(Link)>
                                                </styled.li>
                                                <styled.li>
                                                      <Styled(Link)
                                                            to="/caregiver"
                                                      >
                                                            Caregiver
                                                      </Styled(Link)>
                                                </styled.li>
                                                <styled.li>
                                                      <Styled(Link)
                                                            to="/partnerships"
                                                      >
                                                            Partnerships
                                                      </Styled(Link)>
                                                </styled.li>
                                          </styled.ul>
                                    </styled.div>
                              </Styled(ei)>
                        </Styled($t)>
                        <Styled($t)
                              onMouseEnter=""
                              onMouseLeave=""
                        >
                              <Styled(ei)
                                    activeClassName="active"
                                    tag={[Function]}
                                    to="/how-it-works"
                              >
                                    How it works
                                    
                              </Styled(ei)>
                        </Styled($t)>
                        <Styled($t)
                              onMouseEnter=""
                              onMouseLeave=""
                        >
                              <Styled(ei)
                                    activeClassName="active"
                                    tag={[Function]}
                                    to="/shop"
                              >
                                    Shop
                                    
                              </Styled(ei)>
                        </Styled($t)>
                        <Styled($t)
                              onMouseEnter=""
                              onMouseLeave=""
                        >
                              <Styled(ei)
                                    activeClassName="active"
                                    tag={[Function]}
                                    to="/account"
                              >
                                    Account
                                    
                              </Styled(ei)>
                        </Styled($t)>
                  </Gt>
            </Styled(ei)>
      </gt>
</Styled(jt)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TopMenu
        menuItems={
                Array [
                        Object {
                          "dropdown": Array [
                            Object {
                              "name": "Resident",
                              "route": "/resident",
                            },
                            Object {
                              "name": "Caregiver",
                              "route": "/caregiver",
                            },
                            Object {
                              "name": "Partnerships",
                              "route": "/partnerships",
                            },
                          ],
                          "name": "Who its for",
                          "route": false,
                        },
                        Object {
                          "dropdown": false,
                          "name": "How it works",
                          "route": "/how-it-works",
                        },
                        Object {
                          "name": "Shop",
                          "route": "/shop",
                        },
                        Object {
                          "dropdown": false,
                          "name": "Account",
                          "route": "/account",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TopMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleHover": [Function],
        "props": Object {
          "menuItems": Array [
            Object {
              "dropdown": Array [
                Object {
                  "name": "Resident",
                  "route": "/resident",
                },
                Object {
                  "name": "Caregiver",
                  "route": "/caregiver",
                },
                Object {
                  "name": "Partnerships",
                  "route": "/partnerships",
                },
              ],
              "name": "Who its for",
              "route": false,
            },
            Object {
              "dropdown": false,
              "name": "How it works",
              "route": "/how-it-works",
            },
            Object {
              "name": "Shop",
              "route": "/shop",
            },
            Object {
              "dropdown": false,
              "name": "Account",
              "route": "/account",
            },
          ],
        },
        "refs": Object {},
        "state": Object {
          "dropdownHover": false,
          "isOpen": false,
        },
        "toggle": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(jt)
          light={true}
          toggleable="md"
>
          <gt
                    tag="div"
          >
                    <Styled(Wt)
                              onClick={[Function]}
                              right={true}
                    />
                    <St
                              tag={[Function]}
                              to="/"
                    >
                              <img
                                        alt="HoneyCo Homes"
                                        height={46}
                                        src="IMAGE_MOCK"
                                        width={215}
                              />
                    </St>
                    <Styled(ei)
                              isOpen={false}
                              navbar={true}
                    >
                              <Gt
                                        className="ml-2"
                                        navbar={true}
                                        tag="ul"
                              >
                                        <Styled($t)
                                                  onMouseEnter={[Function]}
                                                  onMouseLeave={[Function]}
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="#"
                                                  >
                                                            Who its for
                                                            <styled.div
                                                                      display="none"
                                                            >
                                                                      <styled.ul>
                                                                                <styled.li>
                                                                                          <Styled(Link)
                                                                                                    to="/resident"
                                                                                          >
                                                                                                    Resident
                                                                                          </Styled(Link)>
                                                                                </styled.li>
                                                                                <styled.li>
                                                                                          <Styled(Link)
                                                                                                    to="/caregiver"
                                                                                          >
                                                                                                    Caregiver
                                                                                          </Styled(Link)>
                                                                                </styled.li>
                                                                                <styled.li>
                                                                                          <Styled(Link)
                                                                                                    to="/partnerships"
                                                                                          >
                                                                                                    Partnerships
                                                                                          </Styled(Link)>
                                                                                </styled.li>
                                                                      </styled.ul>
                                                            </styled.div>
                                                  </Styled(ei)>
                                        </Styled($t)>
                                        <Styled($t)
                                                  onMouseEnter=""
                                                  onMouseLeave=""
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="/how-it-works"
                                                  >
                                                            How it works
                                                            
                                                  </Styled(ei)>
                                        </Styled($t)>
                                        <Styled($t)
                                                  onMouseEnter=""
                                                  onMouseLeave=""
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="/shop"
                                                  >
                                                            Shop
                                                            
                                                  </Styled(ei)>
                                        </Styled($t)>
                                        <Styled($t)
                                                  onMouseEnter=""
                                                  onMouseLeave=""
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="/account"
                                                  >
                                                            Account
                                                            
                                                  </Styled(ei)>
                                        </Styled($t)>
                              </Gt>
                    </Styled(ei)>
          </gt>
</Styled(jt)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(jt)
          light={true}
          toggleable="md"
>
          <gt
                    tag="div"
          >
                    <Styled(Wt)
                              onClick={[Function]}
                              right={true}
                    />
                    <St
                              tag={[Function]}
                              to="/"
                    >
                              <img
                                        alt="HoneyCo Homes"
                                        height={46}
                                        src="IMAGE_MOCK"
                                        width={215}
                              />
                    </St>
                    <Styled(ei)
                              isOpen={false}
                              navbar={true}
                    >
                              <Gt
                                        className="ml-2"
                                        navbar={true}
                                        tag="ul"
                              >
                                        <Styled($t)
                                                  onMouseEnter={[Function]}
                                                  onMouseLeave={[Function]}
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="#"
                                                  >
                                                            Who its for
                                                            <styled.div
                                                                      display="none"
                                                            >
                                                                      <styled.ul>
                                                                                <styled.li>
                                                                                          <Styled(Link)
                                                                                                    to="/resident"
                                                                                          >
                                                                                                    Resident
                                                                                          </Styled(Link)>
                                                                                </styled.li>
                                                                                <styled.li>
                                                                                          <Styled(Link)
                                                                                                    to="/caregiver"
                                                                                          >
                                                                                                    Caregiver
                                                                                          </Styled(Link)>
                                                                                </styled.li>
                                                                                <styled.li>
                                                                                          <Styled(Link)
                                                                                                    to="/partnerships"
                                                                                          >
                                                                                                    Partnerships
                                                                                          </Styled(Link)>
                                                                                </styled.li>
                                                                      </styled.ul>
                                                            </styled.div>
                                                  </Styled(ei)>
                                        </Styled($t)>
                                        <Styled($t)
                                                  onMouseEnter=""
                                                  onMouseLeave=""
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="/how-it-works"
                                                  >
                                                            How it works
                                                            
                                                  </Styled(ei)>
                                        </Styled($t)>
                                        <Styled($t)
                                                  onMouseEnter=""
                                                  onMouseLeave=""
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="/shop"
                                                  >
                                                            Shop
                                                            
                                                  </Styled(ei)>
                                        </Styled($t)>
                                        <Styled($t)
                                                  onMouseEnter=""
                                                  onMouseLeave=""
                                        >
                                                  <Styled(ei)
                                                            activeClassName="active"
                                                            tag={[Function]}
                                                            to="/account"
                                                  >
                                                            Account
                                                            
                                                  </Styled(ei)>
                                        </Styled($t)>
                              </Gt>
                    </Styled(ei)>
          </gt>
</Styled(jt)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TopMenu
    menuItems={
        Array [
            Object {
              "dropdown": Array [
                Object {
                  "name": "Resident",
                  "route": "/resident",
                },
                Object {
                  "name": "Caregiver",
                  "route": "/caregiver",
                },
                Object {
                  "name": "Partnerships",
                  "route": "/partnerships",
                },
              ],
              "name": "Who its for",
              "route": false,
            },
            Object {
              "dropdown": false,
              "name": "How it works",
              "route": "/how-it-works",
            },
            Object {
              "name": "Shop",
              "route": "/shop",
            },
            Object {
              "dropdown": false,
              "name": "Account",
              "route": "/account",
            },
          ]
    }
/>,
}
`;
